### PR TITLE
Add 'Delete All' button.

### PR DESCRIPTION
**Pull request by dragonfax**
Finally, lets add the "delete all" button. This lets @dawbs cancel the workflow and cleanup, or restart the whole thing from scratch if needbe.

/cc @zendesk/samson @grosser 

### Screenshot

samson

### References
 - Jira link: https://zendesk.atlassian.net/browse/TOOLS-957

### Risks
- Level: None

### CODE REVIEW

**Comment by dragonfax** [comment]
Apparently I don't have access to code climate anymore.

**Comment by grosser** [comment]
screen shot 2016-10-26 at 2 52 27 pm

**Comment by grosser** [comment]
... we should remove code-climate and add some tool to do it's job ...

File: app/controllers/admin/deploy_groups_controller.rb [hunk 1]
```diff
@@ -132,6 +145,17 @@ def merge_stage(stage)
     nil
   end
 
+  def delete_stage(stage)
+    return "has no template stage" unless stage.template_stage
```
Comment (grosser) [inline app/controllers/admin/deploy_groups_controller.rb]: extract and reuse ?

File: app/views/admin/deploy_groups/show.html.erb [hunk 2]
```diff
@@ -80,7 +80,18 @@
                                "Merge #{@deploy_group.stages.cloned.count} cloned stages now?"
                     }
         %>
-    <% end %>
+        <%= link_to "Delete Cloned Stages",
+                    delete_all_stages_admin_deploy_group_path(@deploy_group),
+                    class: "btn btn-default #{ 'disabled' unless @deploy_group.stages.cloned.exists? }",
```
Comment (grosser) [inline app/views/admin/deploy_groups/show.html.erb]: doing a count above, so do the count once and then reuse it in these 2 places ?

**Comment by grosser** [comment]
feature looks good!

**Comment by dragonfax** [comment]
@grosser

**Comment by dragonfax** [comment]
That line of code is not highly complex.

File: app/controllers/admin/deploy_groups_controller.rb (hunk 1 above)
Comment (grosser) [inline app/controllers/admin/deploy_groups_controller.rb]: are these not the same checks as in merge ?

**Comment by dragonfax** [comment]
@grosser

**Comment by grosser** [comment]
👍

**Comment by dragonfax** [comment]
@grosser what do you think of code climates complaint?

**Pull request merged**